lib/linux_admin/network_interface.rb

```diff
@@ -20,6 +20,23 @@ def self.dist_class(clear_cache = false)
       end
     end
 
+    def self.list
+      ip_link
+        .map { |net| net["ifname"] }
+        .map { |iface| new(iface) }
+    rescue AwesomeSpawn::CommandResultError => e
+      raise NetworkInterfaceError.new(e.message, e.result)
+    end
+
+    private_class_method def self.ip_link
+      require "json"
+
+      ip_link = Common.run!(Common.cmd("ip"), :params => ["--json", "link"])
```

Comment:
NOTE I think the `--json` output format would be helpful for e.g. `ip addr show` also which currently does some regex parsing of the output but I wanted to leave that for a standalone PR

Reply:
https://github.com/ManageIQ/linux_admin/pull/251